### PR TITLE
Update changelog from 6.1.1 release branch

### PR DESCRIPTION
**Pull request by mdickinson**
This PR brings changelog updates from the 6.1.1 release branch back into the master branch.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
 
+Release 6.1.1
+-------------
+
+Released: 2020-07-23
+
+Traits 6.1.1 is a bugfix release fixing a handful of minor documentation and
+test-related issues with the Traits 6.1.0 release. There are no API-breaking
+changes in this release. It's recommended that all users of Traits 6.1.0
+upgrade to Traits 6.1.1.
```
Comment (kitchoi) [inline CHANGES.rst:12]: Is it really important for users to upgrade?

Comment (mdickinson) [reply CHANGES.rst:12]: Good point. No, not really.

But this text is already in the release changelog, so it doesn't make sense to change it here.

Comment (kitchoi) [reply CHANGES.rst:12]: Ah. OK. If we skip this sentence in the announcement that would be just as good :)

Comment (mdickinson) [reply CHANGES.rst:12]: Yep, I'll do that. Thanks!

**Pull request merged**